### PR TITLE
LP-2726 Tech improvements

### PR DESCRIPTION
**Pull request by asadali145**
[LP-2726](https://philanthropyu.atlassian.net/browse/LP-2726)

1.  Converted `adg/common/lib` to a python package by adding `__init__.py` file.
2.  Created a base serializer for Education, Work Experience and Reference to add `user_application` while creating and updated objects.
3.  Updated docstrings for applications APIs.
4.  Moved webinars API to newly added `webinars/api` package and updated urls accordingly.

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2151/workflows/workflows_id/jobs/2151/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/api/serializers.py (100%)
openedx/adg/lms/applications/api/tests/test_serializers.py (100%)
openedx/adg/lms/webinars/api/tests/test_api_views.py (100%)
openedx/adg/lms/webinars/api/urls.py (100%)
openedx/adg/lms/webinars/api/views.py (100%)
-------------
Total:   34 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2155/workflows/workflows_id/jobs/2155/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/api/serializers.py (94.4%): Missing lines 32
openedx/adg/lms/applications/api/tests/test_serializers.py (100%)
openedx/adg/lms/webinars/api/tests/test_api_views.py (100%)
openedx/adg/lms/webinars/api/urls.py (100%)
openedx/adg/lms/webinars/api/views.py (100%)
-------------
Total:   38 lines
Missing: 1 line
Coverage: 97%
-------------
```

File: openedx/adg/lms/applications/api/serializers.py [hunk 1]
```diff
@@ -8,7 +8,32 @@
...
+
+        Returns:
+            dict: Validated data
+        """
+        user = self.context['request'].user
+
+        if hasattr(user, 'application'):
+            attrs['user_application'] = user.application
+        else:
+            raise serializers.ValidationError('User has no written application')
```
Comment (harris-irfan) [inline openedx/adg/lms/applications/api/serializers.py:32]: For future reference, a good practice to add a blank line before return statement for better readability.

**Review by harris-irfan — APPROVED** [review]
LGTM, Well done! 👏

**Review by javeriaejaz-arbisoft — APPROVED** [review]

**Pull request merged**